### DISCUSSION TITLE
[GSoC] 8. Add time-series anomaly detection support to Anomalib

### DISCUSSION THREAD

**Discussion by samet-akcay** [openvinotoolkit/openvino#16264] [Google Summer of Code]
# Short description
Anomalib currently only contains visual anomaly detection algorithms that operate in the image- and video domain. However, the principles of anomaly detection can be applied to other domains, such as audio, as well. The goal of this project is to add support for anomaly detection in 1D (time-series) data, such as audio signals. The following components would need to be added/modified to achieve time-series anomaly detection support:

Pytorch-Lightning compatible dataset adapters for reading 1D-data
At least 1 fully functional time-series anomaly detection model
Metrics and visualization utilities for qualitative and quantitative evaluation of the model’s performance.
Expected outcomes: Time-series data adapters, 1D anomaly detection model, metrics and visualization utilities

# Skills required/preferred
Basic ML knowledge, Signal processing basics, Python

# Mentors
Dick Ameln, Samet Akcay

# Size of project
175 hours

Difficulty
Medium

**Comment by Ravindu987** [comment]
Hi
I'm expecting to apply for GSoC '23 with OpenVINO and I'm interested in this particular project. I've done the prerequisite task and did the PR.
I've read [this](https://medium.com/openvino-toolkit/how-to-detect-defects-with-an-imbalanced-dataset-63460c177c8b) article on Anomalib and I'd like to gain a deeper understanding. Could you please provide me with some materials
